presto-raptor/src/test/java/com/facebook/presto/raptor/TestRaptorDistributedQueries.java

```diff
@@ -145,4 +150,86 @@ public void testCreateBucketedTable()
 
         assertUpdate("DROP TABLE orders_bucketed");
     }
+
+    @Test
+    public void testTablesSystemTable()
+    {
+        assertUpdate("" +
+                "CREATE TABLE system_tables_none(t1 timestamp, o1 varchar, o2 double, b1 bigint, b2 bigint) " +
+                "WITH (temporal_column = 't1')");
+        assertUpdate("" +
+                "CREATE TABLE system_tables_ordered(t3 timestamp, o3 varchar, o4 double, b3 bigint, b4 bigint) " +
+                "WITH (temporal_column = 't3', ordering = ARRAY['o3', 'o4'])");
+        assertUpdate("" +
+                "CREATE TABLE system_tables_temporal_bucketed(t5 timestamp, o5 varchar, o6 double, b5 bigint, b6 bigint) " +
+                "WITH (temporal_column = 't5', bucket_count = 40, bucketed_on = ARRAY ['b5', 'b6'])");
+        assertUpdate("" +
+                "CREATE TABLE system_tables_temporal_ordered_bucketed(t7 timestamp, o7 varchar, o8 double, b7 bigint, b8 bigint) " +
+                "WITH (temporal_column = 't7', ordering = ARRAY['o7', 'o8'], distribution_name = 'test_distribution', bucket_count = 50, bucketed_on = ARRAY ['b7', 'b8'])");
+
+        MaterializedResult actualResults = computeActual("SELECT * FROM system.tables");
+        assertEquals(actualResults.getTypes(), ImmutableList.of(VARCHAR, VARCHAR, VARCHAR, new ArrayType(VARCHAR), VARCHAR, BIGINT, new ArrayType(VARCHAR)));
+        Map<String, MaterializedRow> map = actualResults.getMaterializedRows().stream()
+                .filter(row -> ((String) row.getField(1)).startsWith("system_tables_"))
+                .collect(ImmutableCollectors.toImmutableMap(row -> ((String) row.getField(1))));
+        assertEquals(map.size(), 4);
+        assertEquals(
+                map.get("system_tables_none").getFields(),
+                asList("tpch", "system_tables_none", "t1", ImmutableList.of(), null, 0L, ImmutableList.of()));
```

Comment:
It seems inconsistent that the arrays are empty, but the other fields are null. I'm thinking we should use null for the arrays as well.